### PR TITLE
refactoring live tests for coupons

### PR DESCRIPTION
**Pull request by alexmamonchik**
Fix commet from @mindeavor

### CODE REVIEW

File: lib/stripe_mock/test_strategies/base.rb [hunk 1]
```diff
@@ -21,6 +21,37 @@ def generate_card_token(card_params={})
...
+          currency: 'usd',
+          max_redemptions: 100,
+          metadata: {
+            created_by: 'admin_acct_1'
+          },
+          duration: 'once'
+        }.merge(params)
+      end
+
+      def create_coupon_percent_of_params(params = {})
```
Comment (mindeavor) [inline lib/stripe_mock/test_strategies/base.rb]: Where does this get used?

Comment (alexmamonchik) [inline lib/stripe_mock/test_strategies/base.rb]: [here] (https://github.com/rebelidealist/stripe-ruby-mock/blob/master/lib/stripe_mock/test_strategies/live.rb#L21). But that method
```ruby
def create_coupon(params={})
  delete_coupon create_coupon_params(params)[:id]
  super
end
```
allows create coupon with `amount_off` only. I have method `create_coupon_percent_of_params` and I going to use it in next commits. Also I found strange behavior when add `percent_off` coupon to subscription. That is not confirmed bug yet.

Comment (mindeavor) [inline lib/stripe_mock/test_strategies/base.rb]: Sorry, I wasn't specific. I do see that `create_coupon_params` gets used, but I couldn't find where `create_coupon_percent_of_params` got used / tested. Maybe just forgot to write a test for it?

Comment (alexmamonchik) [inline lib/stripe_mock/test_strategies/base.rb]: I wanted to add it in another commit because that isn't refactoring, that's new fix/feature.

**Pull request merged**